scripts/mbedtls_framework/collect_test_cases.py

```diff
@@ -126,9 +126,16 @@ def collect_from_script(self, script_name):
     @staticmethod
     def collect_test_directories():
         """Get the relative path for the TLS and Crypto test directories."""
-        mbedtls_root = build_tree.guess_mbedtls_root()
-        directories = [os.path.join(mbedtls_root, 'tests'),
-                       os.path.join(mbedtls_root, 'tf-psa-crypto', 'tests')]
+        #mbedtls_root = build_tree.guess_project_root()
+        project_root = os.getcwd()
+        if build_tree.looks_like_mbedtls_root(project_root):
+            directories = [os.path.join(project_root, 'tests'),
+                           os.path.join(project_root, 'tf-psa-crypto', 'tests')]
+        elif build_tree.looks_like_tf_psa_crypto_root(project_root):
+            directories = [os.path.join(project_root, 'tests')]
+        else:
+            raise Exception("This script must be run from Mbed TLS or TF-PSA-Crypto root")
+
```

Comment:
```suggestion
        if build_tree.is_mbedtls_4_x():
            directories = [os.path.join(project_root, 'tests'),
                           os.path.join(project_root, 'tf-psa-crypto', 'tests')]
        else:
            directories = [os.path.join(project_root, 'tests')]
```
You would need to add `is_mbedtls_4_x()` in build_tree.py, similar to the `is_mbedtls_3_6()` one except that 0x0306 should be replaced by 0x04.
This utility function is missing and I wanted to add it for some time now.

Reply:
Fixed, thanks.

Reply:
I don't see the "You would need to add is_mbedtls_4_x() in build_tree.py" part.

Reply:
Sorry forgot to add the file.